manifest.go

```diff
@@ -67,8 +67,13 @@ func validateManifest(s string) ([]error, error) {
 					for key, value := range v.(map[string]interface{}) {
 						// Check if the key is valid
 						switch key {
-						case "name", "branch", "revision", "version", "source":
+						case "name", "branch", "version", "source":
 							// valid key
+						case "revision":
+							// Check if sha1 hash is abbreviated
+							if valueStr, ok := value.(string); ok && len(valueStr) < 40 {
```

Comment:
This assumes that revisions are only hg/git (which are both commit hashes) but dep also supports other version control systems, such as bazaar. Let's add a test with bazaar revision id's as well to make sure that those won't be affected by this new validation.